### PR TITLE
Subspace v1

### PR DESCRIPTION
**Pull request by nazar-pc**
Please take a look whether you see any major issues.

CI is happy, will start testing in monorepo now.

### CODE REVIEW

**Comment by nazar-pc** [comment]
This PR is not meant to be merged, this is just for review purposes

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    const SEED: [u8; 32] = [
+        35, 2, 52, 4, 51, 55, 23, 84, 91, 10, 111, 12, 13, 222, 151, 16, 228, 211, 254, 45, 92,
+        198, 204, 10, 9, 10, 11, 129, 139, 171, 15, 23,
+    ];
+
+    #[test]
+    fn basic() {
+        let table = Table::generate(&SEED);
+
+        assert!(table.find_quality(0).is_none());
```
Comment (dariolina) [inline src/lib.rs:177]: A minor question: does quality/proof never exist for `challenge_index = 0`, or is it a coincidence? I'd add a note to the spec to have `start_index` and s-bucket numeration start at 1 or be one-off

Comment (nazar-pc) [reply src/lib.rs:177]: Have not tried other seeds, why does it matter?

Comment (dariolina) [reply src/lib.rs:177]: Pedantry, mostly. If the table never has an entry at 0, it makes no sense to query for 0 for every record. According to [design doc](https://github.com/Chia-Network/chiapos/blob/main/documents/proof_of_space.md#full-algorithm) it should exist. This is not a concern for this PR just some minor thing to check.

**Comment by dariolina** [comment]
Looks good to me functionality-wise

**Pull request closed without merging**